### PR TITLE
Second attempt at preventing zombies

### PR DESCRIPTION
**Pull request by zeripath**
* Ensure that the pipes are closed in ssh.go
* Ensure that a cancellable context is passed up in cmd/* http requests
* Make cmd.fail return properly so defers are obeyed
* Ensure that something is sent to stdout in case of blocks here

Follow-up #16314 

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

File: cmd/serv.go [hunk 1]
```diff
@@ -75,26 +73,33 @@ var (
 	alphaDashDotPattern = regexp.MustCompile(`[^\w-\.]`)
 )
 
-func fail(userMessage, logMessage string, args ...interface{}) {
+func fail(userMessage, logMessage string, args ...interface{}) error {
+	fmt.Fprintln(os.Stdout, "")
```
Comment (a1012112796) [inline cmd/serv.go:78]: Why print?

File: integrations/sqlite.ini.tmpl [hunk 2]
```diff
@@ -1,4 +1,4 @@
-APP_NAME = Gitea: Git with a cup of tea
+	APP_NAME = Gitea: Git with a cup of tea
```
Comment (a1012112796) [inline integrations/sqlite.ini.tmpl]:
```suggestion
APP_NAME = Gitea: Git with a cup of tea
```

File: cmd/serv.go [hunk 3]
```diff
@@ -75,26 +73,33 @@ var (
...
 }
 
 func runServ(c *cli.Context) error {
+	ctx, cancel := installSignals()
+	defer cancel()
+
 	// FIXME: This needs to internationalised
 	setup("serv.log", c.Bool("debug"))
 
+	_ = private.SSHLog(context.Background(), true, fmt.Sprintf("Serv: %v", c.Args()))
```
Comment (a1012112796) [inline cmd/serv.go]:
```suggestion
	_ = private.SSHLog(context.Background(), false, fmt.Sprintf("Serv: %v", c.Args()))
```

isErr?

File: cmd/serv.go (hunk 1 above)
Comment (zeripath) [reply cmd/serv.go:78]: Not sure if this is go specific or not but there appears to be a chance to cause a zombie process and failure to read the Exit status if nothing is outputted on stdout.

**Comment by zeripath** [comment]
Please can this get some reviews as it's holding up 1.15 now

File: cmd/serv.go [hunk 4]
```diff
@@ -75,23 +72,28 @@ var (
 	alphaDashDotPattern = regexp.MustCompile(`[^\w-\.]`)
 )
 
-func fail(userMessage, logMessage string, args ...interface{}) {
+func fail(userMessage, logMessage string, args ...interface{}) error {
+	fmt.Fprintln(os.Stdout, "")
```
Comment (6543) [inline cmd/serv.go:78]:
```suggestion
	// There appears to be a chance to cause a zombie process and failure to read the Exit status
	// if nothing is outputted on stdout.
	fmt.Fprintln(os.Stdout, "")
```

**Review by 6543 — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

**Comment by lafriks** [comment]
🚀

**Pull request merged**